core/src/translate/mod.rs

```diff
@@ -97,30 +97,109 @@ pub fn translate(sql_statement: &SqlStatement) -> Result<Statement> {
             comment,
             ..
         } => {
-            let columns = columns
+            let mut primary_key: Vec<usize> = Vec::new();
```

Comment:
how about checking the `constraints` first then columns?

if primary key constraint exists in the parsed schema, then we can assume that there will be no single column primary key declared in the list of sqlparser::ast::column_def.
then I think that the codes can be easier so we may not need comments for extra explanation.

Reply:
I do not think we can assume that, we would need to check and raise an adequate error if such a thing happens. I can clean up the code more though.

Reply:
I have now cleaned up more the code and handled more corner cases that I had not covered earlier, such as multiple PRIMARY KEY definition in the column and mixed definitions in the columns and constraints. I have not swapped the columns and the constraints as the validation must happen for both, in any order.